### PR TITLE
Relock Dependencies

### PR DESCRIPTION
**Pull request by KPrasch**
Currently prevents new CI workflows from running.

See <https://circleci.com/gh/nucypher/nucypher/46185>

### CODE REVIEW

File: scripts/installation/relock_dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+echo "Installing nucypher into new virtual enviorment..."
+pipenv run pip install -e . -r dev-$PREFIX.txt
+
+echo "Checking nucypher CLI entry point..."
+pipenv run nucypher --version
+
+echo "Committing Result..."
+git add Pipfile Pipfile.lock $PREFIX.txt dev-$PREFIX.txt
+git commit -m "Relock dependencies"
```
Comment (derekpierre) [inline scripts/installation/relock_dependencies.sh]: I guess this is mainly for us to run. Although - how does this work when we want to add a dependency as part of a broader change - the "Relock dependencies" commit would be independently committed. Perhaps we could add a 'commit' option, whose default is to not commit?

Comment (KPrasch) [reply scripts/installation/relock_dependencies.sh]: Hmm - how about we always try to commit `Pipfile` as well?

Comment (derekpierre) [reply scripts/installation/relock_dependencies.sh]: If I understand correctly, running the script performs a git commit at the end, and with a comment just about relocking the dependency. If I run this before committing my code, then my code changes also get committed as part of the "relock dependencies" commit too, no?

Comment (vepkenez) [reply scripts/installation/relock_dependencies.sh]: That kind of thing gets annoying.

Comment (KPrasch) [reply scripts/installation/relock_dependencies.sh]: Well, it only runs `git add` against the 4 dependency filepaths, so for you to accidentally include other changes, you'd have to stage them first.

Comment (vepkenez) [reply scripts/installation/relock_dependencies.sh]: The current flow of validating that the requirements match should work fine.  And if your PR fails the reqs validation on circle, then you'd need to relock and push a new commit which is normal and fine.

Comment (KPrasch) [reply scripts/installation/relock_dependencies.sh]: I kept the original script in-tack, for circleCI to use, and relocated it to `scripts/circle`

Comment (vepkenez) [reply scripts/installation/relock_dependencies.sh]: yeah.  It all looks good to me.

Comment (derekpierre) [reply scripts/installation/relock_dependencies.sh]: I still feel funny about it, but I guess we can see how it plays out.

Comment (KPrasch) [reply scripts/installation/relock_dependencies.sh]: I am drawing a bit of inspiration from `bumpversion` - which performs a commit by default - but with a check for staged changes.   I think- it's a good follow up change to include one as well as support cache filepaths on other operating systems.  For now I've been holding on to that script for a while and need to get it off my local tree and am glad to see it merged. :-)

Comment (KPrasch) [reply scripts/installation/relock_dependencies.sh]: I think you are highlighting my critique via this PR - I dont see any reason to ever push a commit that includes both logic and dependency updates in our architecture.

**Pull request merged**